tests/src/3d/sandbox/qgis_3d_sandbox.cpp

```diff
@@ -38,7 +38,11 @@ void initCanvas3D( Qgs3DMapCanvas *canvas )
   QgsMapSettings ms;
   ms.setDestinationCrs( QgsProject::instance()->crs() );
   ms.setLayers( visibleLayers );
-  const QgsRectangle fullExtent = ms.fullExtent();
+  QgsRectangle fullExtent;
+  if ( visibleLayers.isEmpty() )
+    fullExtent = ms.fullExtent();
+  else
+    fullExtent = visibleLayers.at( 0 )->extent(); // focus on 1st layer
```

Comment:
I just to avoid to have the full extends at the load of the project when the main object of interest is not the background (as OSM for example) but other smaller objects in the upper layers.